### ISSUE TITLE
Exception: System.IO.IOException: The parameter is incorrect.

### ISSUE THREAD

**Issue by MahmoodShehab** [PowerShell/PSReadLine#1011]
PS C:\>
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new

-----------------------------------------------------------------------
Last 25 Keys:
 LeftArrow c d ' Enter
 Enter
 c d \ Enter
 Enter
 Enter
 Ctrl+c c d ' Enter
 Enter
 Ctrl+c c d \ Enter
 Ctrl+l Ctrl+l

Exception:
```
System.IO.IOException: The parameter is incorrect.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.Console.SetWindowPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
```

-----------------------------------------------------------------------
![2019-08-27 07_49_25-Administrator_ Windows PowerShell](https://user-images.githubusercontent.com/30172484/63741692-c1018100-c89e-11e9-80d3-94c93cf15152.png)

### Environment

PS version: 5.1.18362.145
PSReadline version: 2.0.0-beta2
os: 10.0.18362.356 (WinBuild.160101.0800)
PS file version: 10.0.18362.356 (WinBuild.160101.0800)
BufferWidth: 319
BufferHeight: 3000

**Comment by MahmoodShehab** [comment]
PS version: 5.1.18362.145
PSReadline version: 2.0.0-beta2
os: 10.0.18362.356 (WinBuild.160101.0800)
PS file version: 10.0.18362.356 (WinBuild.160101.0800)
BufferWidth: 319
BufferHeight: 3000

**Comment by dipakyadav** [comment]
It happened with me as well!!


Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new

-----------------------------------------------------------------------
Last 189 Keys:
 s s h Enter
 s s h Space - - v e r s i o n Enter
 UpArrow LeftArrow LeftArrow LeftArrow RightArrow RightArrow RightArrow Backspace Backspace Backspace Backspace Backspace Backspace LeftArrow Backspace Enter
 c d Space D o w Tab Enter
 s s h Space - i Space Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace s s h Space - i Space " p r a v e e n n a r w a l d o t c o m e c 2 . p e m " Space u b u n t u @ e c 2 - 1 3 - 2 3 4 - 2 3 9 - 6 5 . a p - s o u t h - 1 . c o m p u t e . a m a z o n a w s . c o m Enter
 UpArrow Enter
 DownArrow DownArrow DownArrow DownArrow DownArrow DownArrow DownArrow DownArrow Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter                                                                                                                                                                                       11,2-9        All
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 Enter
 UpArrow

Exception:
```
System.IO.IOException: The parameter is incorrect.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.Console.SetCursorPosition(Int32 left, Int32 top)
   at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
   at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
   at Microsoft.PowerShell.PSConsoleReadLine.HistoryRecall(Int32 direction)
   at Microsoft.PowerShell.PSConsoleReadLine.PreviousHistory(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
```

**Comment by WSLUser** [comment]
For me this issue exists in ConEmu for pwsh. Regular powershell is just fine in ConEmu. pwsh works just fine outside of ConEmu and can run the script above mentioned above. In ConEmu, I get "handle is invalid" error for that script.

Edit: Having re-installed it and doing "refresh tasks", I can open a new dialog in ConEmu. However, this error still presents itself when nested inside of another session. For example, if I open a powershell session and enter pwsh from there, I get this error. If I use Git bash, then pwsh, I get this error, etc.

**Comment by daxian-dbw** [comment]
I took a closer look today, and found the original issue reported here ([`Ctrl+l` causes exception thrown from `Console.SetWindowPosition`](https://github.com/PowerShell/PSReadLine/issues/1011#issue-485580181)) had already been fixed by https://github.com/PowerShell/PSReadLine/pull/790, which [changed the implementation of `ClearScreen`](https://github.com/PowerShell/PSReadLine/commit/bc40ae6f6b44e2eb4bd8b57810079bfd89bffa72#diff-2e67cc4c3200608ae64f002575006135) from using `Console.ScrollBuffer`, `console.SetCursorPosition` and ` console.SetWindowPosition` to applying the escape sequence `"0x1b[2J"`.

That change was included in 2.0.0-beta4 and newer versions, so for those who ran into this issue, please try out the latest 2.0.0-rc1 from PowerShell Gallery.

--------------------

@dipakyadav The issue you report is different (exception thrown from `PSConsoleReadLine.ReallyRender`). There has been lots of fixes around `ReallyRender` since 2.0.0-beta5, and I believe the issue was fixed too. Please try out the 2.0.0-rc1 from PowerShell Gallery.

---------------------

@WSLUser It's not clear to me what issue you ran into. It sounds to me that you had trouble running the script in https://github.com/PowerShell/PSReadLine/issues/1011#issuecomment-531078691. If that's the case, please open a separate issue.
